src/event-cache/EventCache.ts

```diff
@@ -88,7 +91,11 @@ export class EventCache {
             const session: Session = this.sessionManager.getSession();
             this.sessionManager.incrementSessionEventCount();
 
-            if (this.canRecord(session)) {
+            if (
+                this.canRecord(session) &&
+                this.isEventTypePatternAllowed(type) &&
+                Object.keys(eventData).length !== 0
```

Comment:
Since `eventCache.recordEvent` was only accessible via internal implementations, we would never have to worry about empty data. Now that customers can directly send in data and record "custom" events, it is possible that empty data would be passed in, and I thought the benefit of preventing the ingestion of empty data outweighed the use case of sending empty data.

Reply:
Why are we preventing empty data?

Sometimes we might just want to know how many times one event happened (with no data attached to that event)

Reply:
I admit, while implementing this I was mistaking empty data with null values. I'll be making `{}` as default value if not passed in.